TPS_input.py

```diff
@@ -0,0 +1,148 @@
+    result_data = duplicate_data.iloc[:,0:4]
+    
+    # Create variables for the data in A, B, and C
+    data_A = sheet_list[0][(sheet_list[0]['peptide'].isin(duplicate_data['peptide']))].dropna().reset_index(drop=True)
+    data_B = sheet_list[1][(sheet_list[1]['peptide'].isin(duplicate_data['peptide']))].dropna().reset_index(drop=True)
+    data_C = sheet_list[2][(sheet_list[2]['peptide'].isin(duplicate_data['peptide']))].dropna().reset_index(drop=True)
+    
+    # Add the data from sheets A, B, and C respectively
+    result_data = pd.concat([result_data, data_A.iloc[:,4:12]], axis=1, ignore_index=True)
+    result_data = pd.concat([result_data, data_B.iloc[:,4:12]], axis=1, ignore_index=True)
+    result_data = pd.concat([result_data, data_C.iloc[:,4:12]], axis=1, ignore_index=True)
+    
+    print(result_data)
+    
+    # # Get the data for the stats MulitComparison test
+    array_A = np.asarray(data_A.iloc[:,4:12])
+    array_B = np.asarray(data_B.iloc[:,4:12])
+    array_C = np.asarray(data_C.iloc[:,4:12])
+
+
+    
+    # Stack the dataframes into one
+    df = pd.DataFrame()
+    df_A = pd.DataFrame(array_A)
+    df_B = pd.DataFrame(array_B)
+    df_C = pd.DataFrame(array_C)
+    df = pd.concat([df, df_A], axis=1, ignore_index=True)
+    df = pd.concat([df, df_B], axis=1, ignore_index=True)
+    df = pd.concat([df, df_C], axis=1, ignore_index=True)
+    
+    replicates = df.stack().reset_index()
+    replicates = replicates.rename(columns={'level_0': 'id','level_1': 'replicate', 0:'result'})
+    
+    print(replicates)
+
+    result_data.columns = ['peptide', 'protein', 'gene.name', 'modified.sites', '0min', '2min', '4min', '8min', 
+    '16min', '32min', '64min', '128min','0min', '2min', '4min', '8min', 
+    '16min', '32min', '64min', '128min','0min', '2min', '4min', '8min', 
+    '16min', '32min', '64min', '128min']
+
+    # Calculate medians and log2 fold changes
+    medians = pd.DataFrame()
+    log2_fold_changes = pd.DataFrame()
+    for i in result_data.index:
+        # Calculate medians
+        m0 = result_data.iloc[i:i+1,4:5] + result_data.iloc[i:i+1,12:13]+ result_data.iloc[i:i+1,20:21]
+        m2 = result_data.iloc[i:i+1,5:6] + result_data.iloc[i:i+1,13:14]+ result_data.iloc[i:i+1,21:22]
+        m4 = result_data.iloc[i:i+1,6:7] + result_data.iloc[i:i+1,14:15]+ result_data.iloc[i:i+1,22:23]
+        m8 = result_data.iloc[i:i+1,7:8] + result_data.iloc[i:i+1,15:16]+ result_data.iloc[i:i+1,23:24]
+        m16 = result_data.iloc[i:i+1,8:9] + result_data.iloc[i:i+1,16:17]+ result_data.iloc[i:i+1,24:25]
+        m32 = result_data.iloc[i:i+1,9:10] + result_data.iloc[i:i+1,17:18]+ result_data.iloc[i:i+1,25:26]
+        m64 = result_data.iloc[i:i+1,10:11] + result_data.iloc[i:i+1,18:19]+ result_data.iloc[i:i+1,26:27]
+        m128 = result_data.iloc[i:i+1,11:12] + result_data.iloc[i:i+1,19:20]+ result_data.iloc[i:i+1,27:28]
+        # Add medians to median dataframe
+        medians[0].append(m0, ignore_index=True)
```

Comment:
I get an error at this line.  `medians[0]` throws `KeyError: 0`.  If this runs on your machine, what version of pandas do you have?